ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1488,8 +1488,8 @@ withConnectionManager ConnectionManagerArguments {
                 return (DemoteToColdLocalNoop Nothing
                                               (abstractState $ Known connState))
 
-              InboundIdleState _connId _connThread _handle dataFlow ->
-                assert (dataFlow == Duplex) $
+              InboundIdleState _connId _connThread _handle _dataFlow ->
+                -- assert (dataFlow == Duplex) $
```

Comment:
Why this assertion is disabled?

Reply:
This is written in the commit message, by you:

>  server-test: track when to call unregisterOutboundConnection

> This is not perfect, so we also remove an assertion from
connection-manager.  This assertion is benign and also hit rarely (once
per 100_000 simulations).

Reply:
ok; let's just add a `TODO` there.